common/src/main/java/net/whimxiqal/journey/data/DataVersion.java

```diff
@@ -86,51 +91,124 @@ static DataVersion version() {
   }
 
   static void writeVersion(DataVersion version) {
-    File versionFile = Journey.get().proxy().dataFolder().resolve(VERSION_FILE_NAME).toFile();
-    if (!versionFile.exists()) {
-      boolean created = false;
-      try {
-        created = versionFile.createNewFile();
-      } catch (IOException e) {
-        e.printStackTrace();
+    // Prevent writing the same value multiple times, might want to switch to just using the latest here instead of reading from the DB?.
+    if (version.internalVersion <= readVersion().internalVersion) return;
+
+    switch (Settings.STORAGE_TYPE.getValue()) {
+      case MYSQL -> {
```

Comment:
I'd like to keep the number of times we access the `STORAGE_TYPE` setting and use a switch case across all possibilities to a minimum. Maybe you can use the `DataManager` and store some logic in the initialization method that tracks a class to use for executing migrations such as these. Also, what you have written here is the migration from `V1` to `V2`, but what happens when we make another change to the database schema and create a V3? We should keep each migration separate and well defined.

Reply:
I've made it so you define each migration as a sql file, then  you just store the path  to each sql file in a list.
Then, when checking the version, if the the DB version (minus 1) is less than the size of the list, you know you're behind on version, then repeatedly migrate until you're caught up.

The main issue I could see is complex migrations, so it might be better to make migrations use a class instead of just sql file.